src/views/Collectibles/components/AnniversaryAchievement.tsx

```diff
@@ -0,0 +1,69 @@
+import { Box, Button, Flex, InjectedModalProps, Modal, Text } from '@pancakeswap/uikit'
+import confetti from 'canvas-confetti'
+import { useTranslation } from 'contexts/Localization'
+import { delay } from 'lodash'
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+const AnniversaryImage = styled.img`
+  border-radius: 50%;
+  height: 128px;
+  margin-bottom: 24px;
+  margin-right: 8px;
+  width: 128px;
+`
+
+const showConfetti = () => {
+  confetti({
+    resize: true,
+    particleCount: 200,
+    startVelocity: 30,
+    gravity: 0.5,
+    spread: 350,
+    origin: {
+      x: 0.5,
+      y: 0.3,
+    },
+  })
+}
+
+interface AnniversaryModalProps extends InjectedModalProps {
+  onClick: () => Promise<void>
+}
+
+const AnniversaryAchievementModal: React.FC<AnniversaryModalProps> = ({ onDismiss, onClick }) => {
+  const { t } = useTranslation()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleClick = async () => {
+    setIsLoading(true)
+    try {
+      await onClick()
+    } catch (e) {
+      onDismiss()
+    }
+    onDismiss()
+  }
```

Comment:
Small thing but if we are not doing anything with the error we can simply do this
```
try {
  await onClick()
} finally {
  onDismiss()
}